src/server/response.cpp

```diff
@@ -169,12 +160,29 @@ HTTP400HtmlResponse& HTTP400HtmlResponse::operator+(InvalidUrlMsg /*unused*/)
   return *this + msgTmpl.render({"url", requestUrl});
 }
 
-HTTP400HtmlResponse& HTTP400HtmlResponse::operator+(const std::string& msg)
+HTTP500HtmlResponse::HTTP500HtmlResponse(const InternalServer& server,
+                                         const RequestContext& request)
+  : HTTPErrorHtmlResponse(server,
+                          request,
+                          MHD_HTTP_INTERNAL_SERVER_ERROR,
+                          "Internal Server Error",
+                          "Internal Server Error")
 {
-  m_data["details"].push_back({"p", msg});
-  return *this;
+  // operator+() is a state-modifying operator (akin to operator+=)
+  *this + "An internal server error occured. We are sorry about that :/";
 }
 
+std::unique_ptr<ContentResponse> HTTP500HtmlResponse::generateResponseObject() const
+{
+  // We want a 500 response to be a minimalistic one (so that the server doesn't
+  // have to provide additional resources required for its proper rendering)
+  // ";raw=true" in the MIME-type below disables response decoration
+  // (see ContentResponse::contentDecorationAllowed())
```

Comment:
This is in the same time nice and a bit disturbing.
The "raw=true" has been added to handle something totally different (and on which I disagreed, https://github.com/kiwix/libkiwix/pull/403#issuecomment-671277426) and in the same time, it fits perfectly here ..